internal/bminventory/inventory.go

```diff
@@ -458,6 +458,13 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		monitoredOperators = append(monitoredOperators, newOLMOperators...)
 	}
 
+	platform := params.NewClusterParams.Platform
+	if models.PlatformTypeBaremetal == *platform.Type && *params.NewClusterParams.UserManagedNetworking {
```

Comment:
Note that in line 421 `setDefaultRegisterClusterParams` will set the platform to Baremetal if the user didn't specify it.
Meaning whatever checks you are doing for the params input it should happen before we set defaults.